src/systems/drawing-menu-system.js

```diff
@@ -0,0 +1,200 @@
+      this.camera = this.sceneEl.querySelector("#avatar-pov-node");
+      this.observer.observe(this.sceneEl, { childList: true, attributes: true, subtree: true });
+      this.sceneEl.addEventListener("object3dset", this.setDirty.bind(this));
+      this.sceneEl.addEventListener("object3dremove", this.setDirty.bind(this));
+    });
+  }
+
+  tick() {
+    if (!this.cursorControllers || this.cursorControllers.length === 0 || !this.camera) return;
+
+    if (this.sceneEl.is("frozen")) {
+      if (this.dirty) {
+        this.populateEntities(this.targets);
+        this.dirty = false;
+      }
+
+      const hovered = this.getHovered();
+      if (hovered) {
+        for (let i = 0; i < this.cursorControllers.length; i++) {
+          if (this.cursorControllers[i].components["cursor-controller"].intersection) {
+            const intersectionPoint = this.cursorControllers[i].components["cursor-controller"].intersection.point;
+            this.showMenu(hovered, intersectionPoint);
+          }
+        }
+      }
+      this.handleButtons();
+      this.scaleMenus();
+    }
+  }
+
+  getMenuScale(distance) {
+    if (distance <= 1.5) {
+      return 0.5;
+    }
+    return Math.min(3, distance - 1);
+  }
+
+  scaleMenus = (() => {
+    const cameraWorldPos = new THREE.Vector3();
+    return function() {
+      if (!this.drawingMenus || this.drawingMenus.length === 0) return;
+
+      this.camera.object3D.getWorldPosition(cameraWorldPos);
+
+      for (let i = 0; i < this.drawingMenus.length; i++) {
+        const menu = this.drawingMenus[i];
+        if (menu.object3D.visible && !menu.getAttribute("animation__show")) {
+          const dist = cameraWorldPos.distanceTo(menu.object3D.position);
+          menu.object3D.scale.setScalar(this.getMenuScale(dist));
+          menu.object3D.matrixNeedsUpdate = true;
+        }
+      }
+    };
+  })();
+
+  handleButtons = (() => {
+    const remotes = ["rightRemote", "leftRemote"];
+    return function() {
+      const interaction = AFRAME.scenes[0].systems.interaction;
+      const userinput = AFRAME.scenes[0].systems.userinput;
+
+      for (let i = 0; i < remotes.length; i++) {
+        const remote = remotes[i];
+        if (interaction.state[remote]) {
+          const hovered = interaction.state[remote].hovered;
+          if (
+            hovered &&
+            interaction.options[remote] &&
+            userinput.get(interaction.options[remote].grabPath) &&
+            hovered.components.tags &&
+            hovered.components.tags.data.singleActionButton &&
+            hovered.parentEl.parentEl.components["networked-drawing"]
+          ) {
+            if (hovered.classList.contains("undo-drawing")) {
+              hovered.parentEl.parentEl.components["networked-drawing"].undoDraw();
+            } else if (hovered.classList.contains("delete-drawing")) {
+              NAF.utils.takeOwnership(hovered.parentEl.parentEl);
```

Comment:
Having to crawl up the hierarchy here is fine I think, but might prefer to have a helper to build the association between the hovered entity and the networked entity, so that this code is a bit more robust to changes in the `html` structure. (e.g. if there was another child inserted between the networked el and the hovered el).

Reply:
I opted against doing that because this runs in `tick` (though tbf, only while in pause-mode). As such, doing any kind of hierarchy search, while being more robust, will be significantly more expensive. If you have any ideas of how to do this and still be efficient, please let me know.

Reply:
I think if we do the search once we can re-use the result (rather than redo the search)

Reply:
There's N number of drawings, where N is up to the number of users in the space. This system runs for all of them, not just one drawing.